### ISSUE TITLE
[WAF autoscale] autoscale BIG-IP fails to join cluster

### ISSUE THREAD

**Issue by tkam8** [F5Networks/f5-aws-cloudformation#22]
## Description
Additional autoscaled/secondary BIG-IP fails to join the cluster.  
Seems that by default only "restrictedSrcAddress" is allowed inbound 8443 access in the CFT created security group. The value I input for the restrictedSrcAddress is a /32, but as it is now, only a 0.0.0.0/0 would seem to work.

I am thinking the options are to add another field to input/select the internal subnet for allowing cluster join to occur, OR somehow adding the internal subnet by default in the scripts that creates the secgroup. 

Is this just my environment? 

Repro Steps:
1. Deploy CFT
2. Trigger autoscale
3. Confirm that additional BIG-IP comes up but does not join the cluster

Log of affected BIG-IP
/var/log/custom-config.log
```
2017-07-18T10:28:22.227Z info: /config/cloud/aws/node_modules/f5-cloud-libs/scripts/runScript.js called with /usr/bin/f5-rest-node /config/cloud/aws/node_modules/f5-cloud-libs/scripts/runScript.js --log-level info --file /config/cloud/aws/custom-config.sh --cwd /config/cloud/aws -o /var/log/custom-config.log --wait-for ONBOARD_DONE
2017-07-18T10:28:22.236Z info: Waiting for ONBOARD_DONE
2017-07-18T10:32:56.027Z info: /config/cloud/aws/custom-config.sh starting.
2017-07-18T10:32:56.045Z info: Tue Jul 18 19:32:56 JST 2017
2017-07-18T10:32:56.061Z error: % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2017-07-18T10:32:56.061Z error: Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M  0    44    0    44    0     0   4564      0 --:--:-- --:--:-- --:--:-- 44000
2017-07-18T10:32:57.608Z info: Attempting to Join or Initiate Autoscale Cluster
2017-07-18T10:32:57.759Z info: 2017-07-18T10:32:57.757Z info: /config/cloud/aws/node_modules/f5-cloud-libs/scripts/autoscale.js called with /usr/bin/f5-rest-node /config/cloud/aws/node_modules/f5-cloud-libs/scripts/autoscale.js --cloud aws --provider-options s3Bucket:awsmpautoscalingwafviacft-s3bucket-11zlr5b2w9mjt,mgmtPort:8443 --host localhost --port 8443 --user awsadmin --password-url file:///config/cloud/aws/.adminPassword --device-group autoscale-group --block-sync -c join --log-level info --output /var/log/aws-autoscale.log
2017-07-18T10:32:58.004Z info: 2017-07-18T10:32:58.004Z info: Initializing BIG-IP.
2017-07-18T10:32:58.386Z info: 2017-07-18T10:32:58.386Z info: Getting this instance ID.
2017-07-18T10:32:58.387Z info: 2017-07-18T10:32:58.387Z info: Getting info on all instances.
2017-07-18T10:32:58.812Z info: 2017-07-18T10:32:58.812Z info: Determining master instance id.
2017-07-18T10:32:58.813Z info: 2017-07-18T10:32:58.812Z info: Possible master ID: i-03a409857dd4e
2017-07-18T10:32:58.813Z info: 2017-07-18T10:32:58.813Z info: Valid master ID: i-03a409857dd4e
2017-07-18T10:32:58.814Z info: 2017-07-18T10:32:58.813Z info: Using master ID: i-03a409857dd4e
2017-07-18T10:32:58.814Z info: This instance is not master
2017-07-18T10:32:58.815Z info: 2017-07-18T10:32:58.815Z info: Cluster action JOIN
2017-07-18T10:32:58.886Z info: 2017-07-18T10:32:58.886Z info: Setting config sync IP.
2017-07-18T10:32:59.001Z info: 2017-07-18T10:32:59.000Z info: Joining cluster.
2017-07-18T11:06:17.971Z error: 2017-07-18T11:06:17.970Z error: connect ECONNREFUSED 10.0.11.77:8443   <---!!!!!
2017-07-18T11:06:17.972Z info: 2017-07-18T11:06:17.972Z info: Autoscale finished.
2017-07-18T11:06:17.978Z info: Tue Jul 18 20:06:17 JST 2017
2017-07-18T11:06:17.979Z info: custom-config.sh complete
```

This log told me it is a security group related issue so I modified the CFT created security group to allow inbound 8443 from 10.0.0.0/16, redeployed the BIG-IP and confirmed it succeeded. 

```
----snip----
2017-07-18T13:30:05.227Z info: 2017-07-18T13:30:05.227Z info: Initializing BIG-IP.
2017-07-18T13:30:05.761Z info: 2017-07-18T13:30:05.760Z info: Getting this instance ID.
2017-07-18T13:30:05.761Z info: 2017-07-18T13:30:05.761Z info: Getting info on all instances.
2017-07-18T13:30:06.349Z info: 2017-07-18T13:30:06.348Z info: Determining master instance id.
2017-07-18T13:30:06.356Z info: 2017-07-18T13:30:06.349Z info: Possible master ID: i-03a409857dd4e
2017-07-18T13:30:06.349Z info: Valid master ID: i-03a409857dd4e
2017-07-18T13:30:06.360Z info: 2017-07-18T13:30:06.360Z info: Using master ID: i-03a409857dd4e
2017-07-18T13:30:06.360Z info: 2017-07-18T13:30:06.360Z info: This instance is not master
2017-07-18T13:30:06.361Z info: 2017-07-18T13:30:06.360Z info: Cluster action JOIN
2017-07-18T13:30:06.428Z info: 2017-07-18T13:30:06.428Z info: Setting config sync IP.
2017-07-18T13:30:06.593Z info: 2017-07-18T13:30:06.592Z info: Joining cluster.
2017-07-18T13:30:07.014Z info: 2017-07-18T13:30:07.014Z info: Checking remote host for cluster readiness.
2017-07-18T13:30:07.191Z info: 2017-07-18T13:30:07.142Z info: Getting local hostname for trust.
2017-07-18T13:30:07.191Z info: 2017-07-18T13:30:07.155Z info: Getting local management address.
2017-07-18T13:30:07.163Z info: Adding to remote trust.
2017-07-18T13:30:16.278Z info: 2017-07-18T13:30:16.278Z info: Adding to remote device group.
2017-07-18T13:30:16.725Z info: 2017-07-18T13:30:16.721Z info: Checking for datasync-global-dg.
2017-07-18T13:30:17.137Z info: 2017-07-18T13:30:17.137Z info: Telling remote to sync.
2017-07-18T13:30:47.227Z info: 2017-07-18T13:30:47.226Z info: Telling remote to sync datasync-global-dg request.
2017-07-18T13:30:47.634Z info: 2017-07-18T13:30:47.633Z info: Waiting for sync to complete.
2017-07-18T13:31:18.102Z info: 2017-07-18T13:31:18.101Z info: Sync not yet complete.
2017-07-18T13:31:18.102Z info: 2017-07-18T13:31:18.102Z info: Following recommended action. Syncing group autoscale-group
2017-07-18T13:31:48.189Z info: 2017-07-18T13:31:48.188Z info: Telling remote to sync.
2017-07-18T13:32:18.524Z info: 2017-07-18T13:32:18.523Z info: Telling remote to sync datasync-global-dg request.
2017-07-18T13:32:18.652Z info: 2017-07-18T13:32:18.651Z info: Waiting for sync to complete.
2017-07-18T13:32:29.114Z info: 2017-07-18T13:32:29.113Z info: Sync complete.
2017-07-18T13:32:29.115Z info: 2017-07-18T13:32:29.115Z info: Autoscale finished.
2017-07-18T13:32:29.121Z info: Tue Jul 18 22:32:29 JST 2017
```

## Template
https://github.com/F5Networks/f5-aws-cloudformation/tree/master/supported/solutions/autoscale/waf

## Severity Level
For bugs, enter the bug severity level. Do not set any labels.

Severity: 2

**Comment by kskoh** [comment]
Bug#674703.
Current workaround.  When the security group is created by the CFT, one need to edit it and add another rule with port of management port(default 8443) and source of VPC IPV4 net.  We will fix it such that the CFT will automatically create two rules for the management port, one with source as the user provided Restricted Source Address and another one with source of the peer BIG-IPs.

**Comment by kskoh** [comment]
Resolved in template version 2.5.0. in Release 5 published August 11th, 2017.
